### PR TITLE
update fork for ibc-host

### PR DESCRIPTION
**Pull request by vuong177**
## Description
- I fixed `fork.go` for ibc-host. It must run `RunMigrations` after `icamodule.InitModule`

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by faisalnaveedRNS — APPROVED** [review]

File: app/app.go [hunk 1]
```diff
@@ -581,9 +581,6 @@ func New(
 
 	app.sm.RegisterStoreDecoders()
 
-	// register upgrade
-	app.RegisterUpgradeHandlers(cfg)
```
Comment (catShaark) [inline app/app.go:585]: can you explain this ?

File: app/upgrade/fork.go [hunk 2]
```diff
@@ -58,6 +49,15 @@ func CreateUpgradeHandler(mm *module.Manager, configurator module.Configurator,
 		}
 		icamodule.InitModule(ctx, controllerParams, hostParams)
 
+		newVM, err := mm.RunMigrations(ctx, configurator, vm)
```
Comment (catShaark) [inline app/upgrade/fork.go:52]: legit, icamodule.InitModule should be called before mm.RunMigrations as described in this [doc](https://github.com/cosmos/ibc-go/blob/031f6a39d5f3f778e18faeae3242b1213d2fe9bd/docs/migrations/v2-to-v3.md?plain=1#L52)

**Review by mustafapylons — APPROVED** [review]
Looks good to me.

File: app/upgrade/fork.go (hunk 2 above)
Comment (vuong177) [reply app/upgrade/fork.go:52]: thank @catShaark

File: app/app.go (hunk 1 above)
Comment (vuong177) [reply app/app.go:585]: I add it after. Only for better readable struct.

**Comment by vuong177** [comment]
@faisalnaveedRNS Do you get any problem when upgrade ?

**Review by faddat — APPROVED** [review]

**Pull request merged**